webpack.common.js

```diff
@@ -12,7 +14,8 @@ const gitBranch = require('child_process')
     .execSync('git rev-parse --abbrev-ref HEAD')
     .toString().trim();
 
-module.exports = {
+/** @type {import('webpack').Configuration} */
```

Comment:
@trusktr Can you explain a little what this does? Does it affect the Open MCT build artifact?

Reply:
Not at all. This is a TypeScript JSDoc comment that VS Code understands (thanks to built-in support from TypeScript).

Now if you open this file in VS Code, and down below you modify `entry` to `entryy`, it will show you a red squiggly line telling you that `entryy` is not a valid property for a webpack config object.

Apart from this, it has no effect on existing build processes, it's only for IDE convenience.

Reply:
Try hovering on any of the properties in the config to see a tooltip describing the property.

Reply:
Got it, thanks. Sounds like it will make Webpack config changes easier.

Reply:
We should clarify that this project is not built with TypeScript. The presence of this file may lead developers to conclude that it does. 

Could we please add a comment to this file (TypeScript supports comments in its config apparently, even though they're JSON!) to this effect?

Reply:
> Got it, thanks. Sounds like it will make Webpack config changes easier.

This, plus it covers all JS files in the project (and additionally, if using the Volar extension, all Vue files, although type support is lacking in some areas in Vue files).

More on that here: https://github.com/nasa/openmct/issues/4690